src/python_testing/TC_RVCRUNM_2_2.py

```diff
@@ -0,0 +1,202 @@
+        return "kWaterTankLidOpen(0x46)"
+    elif error_eum == Clusters.RvcRunMode.Enums.ModeTag.kMopCleaningPadMissing:
+        return "kMopCleaningPadMissing(0x47)"
+    elif error_eum == Clusters.RvcRunMode.Enums.ModeTag.kBatteryLow:
+        return "kBatteryLow(0x48)"
+
+
+class TC_RVCRUNM_2_2(MatterBaseTest):
+
+    def __init__(self, *args):
+        super().__init__(args)
+        self.endpoint = 0
+        self.mode_a = 0
+        self.mode_b = 0
+        self.supported_run_modes = {}  # these are the ModeOptionStructs
+        self.supported_run_modes_dut = []
+        self.idle_mode_dut = 0
+
+    async def read_mod_attribute_expect_success(self, cluster, attribute):
+        return await self.read_single_attribute_check_success(
+            endpoint=self.endpoint, cluster=cluster, attribute=attribute)
+
+    async def read_run_supported_modes(self) -> Clusters.Objects.RvcRunMode.Attributes.SupportedModes:
+        ret = await self.read_mod_attribute_expect_success(
+            Clusters.RvcRunMode,
+            Clusters.RvcRunMode.Attributes.SupportedModes)
+        asserts.assert_true(type_matches(ret, Clusters.Objects.RvcRunMode.Attributes.SupportedModes), "")
+        return ret
+
+    async def read_current_mode_with_check(self, expected_mode):
+        run_mode = await self.read_mod_attribute_expect_success(
+            Clusters.RvcRunMode,
+            Clusters.RvcRunMode.Attributes.CurrentMode)
+        asserts.assert_true(run_mode == expected_mode,
+                            "Expected the current mode to be %i, got %i" % (expected_mode, run_mode))
+
+    async def send_change_to_mode_cmd(self, new_mode) -> Clusters.Objects.RvcRunMode.Commands.ChangeToModeResponse:
+        ret = await self.send_single_cmd(cmd=Clusters.Objects.RvcRunMode.Commands.ChangeToMode(newMode=new_mode),
+                                         endpoint=self.endpoint)
+        asserts.assert_true(type_matches(ret, Clusters.Objects.RvcRunMode.Commands.ChangeToModeResponse),
+                            "Unexpected return type for ChangeToMode")
+        return ret
+
+    async def send_change_to_mode_with_check(self, new_mode, expected_error):
+        response = await self.send_change_to_mode_cmd(new_mode)
+        asserts.assert_true(response.status == expected_error,
+                            "Expected a ChangeToMode response status of %s, got %s" %
+                            (error_enum_to_text(expected_error), error_enum_to_text(response.status)))
+
+    # Prints the instruction and waits for a user input to continue
+    def print_instruction(self, step_number, instruction):
+        self.print_step(step_number, instruction)
+        input("Press Enter when done.\n")
+
+    def pics_TC_RVCRUNM_2_2(self) -> list[str]:
+        return ["RVCRUNM.S"]
+
+    @async_test_body
+    async def test_TC_RVCRUNM_2_2(self):
+
+        if 'PIXIT_ENDPOINT' not in self.matter_test_config.global_test_params or \
```

Comment:
as above - use the endpoint on the command line directly to make this easier for the TH